app/src/main/java/com/xmartlabs/taskloans/data/common/DataHelper.kt

```diff
@@ -0,0 +1,20 @@
+package com.xmartlabs.taskloans.data.common
+
+import retrofit2.HttpException
+
+object DataHelper {
+  private const val ERROR_INVALID_USER = 401
+  private const val ERROR_USER_CONFLICT = 409
+
+  suspend fun <T> mapServiceError(errorMapper: (HttpException) -> Exception, serviceCall: suspend () -> T): T = try {
```

Comment:
what about making this an extension of a suspend function? idk if that possible but in that case we can call just
`suspendFunction().mapServiceError(::serviceErrorMapper)` we can even have the error mapper by default